tools/cldr-apps/src/main/java/org/unicode/cldr/web/DataPage.java

```diff
@@ -1371,13 +1370,20 @@ public static CheckCLDR.Options getOptions(CookieSession session, CLDRLocale loc
     private static final boolean DEBUG_DATA_PAGE = false;
     private String creationTime = null; // only used if DEBUG_DATA_PAGE
 
+
+    final private CLDRFile cldrFile;
+    final private STUsersChoice userVoteStatus;
+
     DataPage(PageId pageId, SurveyMain sm, CLDRLocale loc, String prefix, XPathMatcher matcher) {
         this.locale = loc;
         this.sm = sm;
         this.matcher = matcher;
         xpathPrefix = prefix;
         ballotBox = sm.getSTFactory().ballotBoxForLocale(locale);
         this.pageId = pageId;
+        this.cldrFile = sm.getSTFactory().make(locale.getBaseName());
+        cldrFile.setSupplementalDirectory(sm.getSupplementalDirectory());
```

Comment:
Just noticed this. Over the life of the ST, the same supplemental directory should be used for everything; otherwise bad things can happen. I'm surprised that sm.getSTFactory() doesn't already use that factory, and thus cldrFile automatically have it.